uhubctl.c

```diff
@@ -1051,13 +1092,427 @@ static int usb_find_hubs(void)
+        for (int i = 0; i < config->bNumInterfaces; i++) {
+            const struct libusb_interface *interface = &config->interface[i];
+            for (int j = 0; j < interface->num_altsetting; j++) {
+                const struct libusb_interface_descriptor *altsetting = &interface->altsetting[j];
+                if (altsetting->bInterfaceClass == LIBUSB_CLASS_MASS_STORAGE) {
+                    ret = 1;
+                    goto out;
+                }
+            }
+        }
+    out:
+        libusb_free_config_descriptor(config);
+    }
+    return ret;
+}
+
+
+// Helper function to create the status flags JSON object
+cJSON* create_status_flags_json(int port_status)
+{
+    cJSON* flags = cJSON_CreateObject();
+    
+    struct {
+        int mask;
+        const char* name;
+    } flag_defs[] = {
+        {USB_PORT_STAT_CONNECTION, "connection"},
+        {USB_PORT_STAT_ENABLE, "enable"},
+        {USB_PORT_STAT_SUSPEND, "suspend"},
+        {USB_PORT_STAT_OVERCURRENT, "overcurrent"},
+        {USB_PORT_STAT_RESET, "reset"},
+        {USB_PORT_STAT_POWER, "power"},
+        {USB_PORT_STAT_LOW_SPEED, "lowspeed"},
+        {USB_PORT_STAT_HIGH_SPEED, "highspeed"},
+        {USB_PORT_STAT_TEST, "test"},
+        {USB_PORT_STAT_INDICATOR, "indicator"},
+        {0, NULL}
+    };
+    
+    for (int i = 0; flag_defs[i].name != NULL; i++) {
+        cJSON_AddBoolToObject(flags, flag_defs[i].name, (port_status & flag_defs[i].mask) != 0);
+    }
+    
+    return flags;
+}
+
+// Helper function to create human-readable descriptions of set flags
+cJSON* create_human_readable_json(int port_status)
+{
+    cJSON* human_readable = cJSON_CreateObject();
+    
+    struct {
+        int mask;
+        const char* name;
+        const char* description;
+    } flag_defs[] = {
+        {USB_PORT_STAT_CONNECTION, "connection", "Device is connected"},
+        {USB_PORT_STAT_ENABLE, "enable", "Port is enabled"},
+        {USB_PORT_STAT_SUSPEND, "suspend", "Port is suspended"},
+        {USB_PORT_STAT_OVERCURRENT, "overcurrent", "Over-current condition exists"},
+        {USB_PORT_STAT_RESET, "reset", "Port is in reset state"},
+        {USB_PORT_STAT_POWER, "power", "Port power is enabled"},
+        {USB_PORT_STAT_LOW_SPEED, "lowspeed", "Low-speed device attached"},
+        {USB_PORT_STAT_HIGH_SPEED, "highspeed", "High-speed device attached"},
+        {USB_PORT_STAT_TEST, "test", "Port is in test mode"},
+        {USB_PORT_STAT_INDICATOR, "indicator", "Port indicator control"},
+        {0, NULL, NULL}
+    };
+    
+    for (int i = 0; flag_defs[i].name != NULL; i++) {
+        if (port_status & flag_defs[i].mask) {
+            cJSON_AddStringToObject(human_readable, flag_defs[i].name, flag_defs[i].description);
+        }
+    }
+    
+    return human_readable;
+}
+
+// Helper function to determine port speed
+void get_port_speed(int port_status, char** speed_str, int64_t* speed_bits)
```

Comment:
`speed_bits` should be `speed_bps` (bits per second). That should be in emitted json as well.